src/vcs/git.rs

```diff
@@ -192,6 +205,77 @@ impl<'repo> Repository {
         self.commit_to_history(commit)
     }
 
+    /// Get the [`Diff`] between two commits.
+    pub fn diff(&self, from: &'repo git2::Commit, to: &'repo git2::Commit) -> Result<Diff, Error> {
+        use git2::{Delta, Patch};
+
+        let mut diff = Diff::new();
+        let git_diff = self.diff_commits(from, Some(to))?;
+
+        for (idx, delta) in git_diff.deltas().enumerate() {
+            match delta.status() {
+                Delta::Added => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let name = path.file_name().expect("path should have a name");
+                    let parent = path.parent().expect("path should have a parent");
+
+                    let label = file_system::Label::try_from(&*name.to_string_lossy())?;
+                    let parent = file_system::Path::try_from(parent.to_path_buf())?;
+
+                    diff.add_created_file(&label, &parent);
+                },
+                Delta::Deleted => {
+                    let diff_file = delta.old_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let name = path.file_name().expect("path should have a name");
+                    let parent = path.parent().expect("path should have a parent");
+
+                    let label = file_system::Label::try_from(&*name.to_string_lossy())?;
+                    let parent = file_system::Path::try_from(parent.to_path_buf())?;
```

Comment:
This dance should be factored out somewhere, and maybe it can be re-written differently :roll_eyes:

Reply:
I tried `file_system::Path::try_from()`, but it's also pretty painful.

Reply:
What's the pain point here? I'd like to improve it if possible :) `Path` is there to ensure we have `NonEmpty`-ness and making sure user input is sanitised somewhat.